pcsx2/VMManager.cpp

```diff
@@ -1267,10 +1267,46 @@ void VMManager::PrecacheCDVDFile()
 	}
 }
 
-bool VMManager::Initialize(VMBootParameters boot_params)
+void VMManager::InitializeAsync(
+	const VMBootParameters& boot_params,
+	VMBootHardcoreDisableCallback hardcore_disable_callback,
+	VMBootDoneCallback done_callback)
+{
+	Error error;
+	VMBootResult result = VMManager::Initialize(boot_params, &error);
+
+	if (result == VMBootResult::PromptDisableHardcoreMode)
+	{
+		std::string reason;
+		if (DebugInterface::getPauseOnEntry())
```

Comment:
```suggestion
		if (!DebugInterface::getPauseOnEntry())
```

Reply:
I've fixed the flipped around strings. I think this PR is fixing enough problems, I don't want to increase the scope even more with the boot and debug issue. Maybe I could have a look later.

Reply:
> So instead of removing this functionality outright, how about we make it explicit. Replace the harcore mode checkbox with a combo box with three option: Enabled, which makes it impossible to temporarily disable hardcore mode, Always Prompt, which has a similar behaviour as the achievements system had originally (or we could even make it so it prompts when hardcore is to be enabled too), and Disabled.
> 
> The majority of users get a safer experience, the ones that want a terse UI won't have anything to complain about.
> 
> Does anyone agree or is this idea horrible?
> 
> /rant

I'm in favour of the three state combobox.

Reply:
Great. How about wait for a future PR for that then, since it sounds like the code I'm adding will still be needed for that case.

Reply:
You reported an issue about the big picture hardcore mode prompt being shown twice in certain circumstances. I haven't been able to reproduce this, although I saw a place in the code where it could happen, so I added some logic that should fix it.